// Handles routing and listens for routing goals

syntax = "proto3";

package ac;

import "google/protobuf/empty.proto";
import "routine.proto";

service Runtime {
  // Fetches where the current node is
  rpc GetCurrentNode(google.protobuf.Empty) returns (Routine.Node) {}

  rpc SetNode(Routine.Node) returns (google.protobuf.Empty) {}

  rpc GoToNode(Routine.Node) returns (google.protobuf.Empty) {}
}

message RuntimeState {
  // current node
  Routine.Node current_node = 1;

  // desired target node
  Routine.Node target_node = 2;

  // non null if current processing edge (condition/action)
  Routine.Edge current_edge = 3;
}
